src/System.CommandLine.Subsystems/PipelineResult.cs

```diff
@@ -13,6 +16,22 @@ public class PipelineResult(ParseResult? parseResult, string rawInput, Pipeline?
     public bool AlreadyHandled { get; set; }
     public int ExitCode { get; set; }
 
+    public void AddErrors(IEnumerable<ParseError> errors)
+    {
+        if (errors is not null)
+        {
+            this.errors.AddRange(errors);
+        }
+    }
+
+    public void AddError(ParseError error)
+        => errors.Add(error);
+
+    public IEnumerable<ParseError> GetErrors(bool excludeParseErrors = false)
```

Comment:
Returning `ParseError` but an argument called `excludeParseErrors` seems odd - are we going to rename `ParseError`?